languagetool-language-modules/ar/src/test/java/org/languagetool/rules/ar/ArabicMorfologikSpellerRuleTest.java

```diff
@@ -0,0 +1,62 @@
+/* LanguageTool, a natural language style checker
+ * Copyright (C) 2019 Sohaib Afifi, Taha Zerrouki
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+package org.languagetool.rules.ar;
+
+import org.junit.Test;
+import org.languagetool.JLanguageTool;
+import org.languagetool.Languages;
+import org.languagetool.TestTools;
+import org.languagetool.language.Arabic;
+import org.languagetool.rules.spelling.morfologik.MorfologikSpeller;
+
+import java.io.IOException;
+
+import static org.junit.Assert.*;
+
+public class ArabicMorfologikSpellerRuleTest {
+
+  @Test
+  public void testIsMisspelledFromDict() throws IOException {
+    MorfologikSpeller speller = new MorfologikSpeller("/ar/arabic.dict");
+
+    // The dict is not considering the ta3reef
+    // assertFalse(speller.isMisspelled("السلام عليكم"));
+
+    assertTrue(speller.isMisspelled("علييكم"));
+
+    assertTrue(speller.isMisspelled("العباره"));
+  }
+
+  @Test
+  public void testIsMisspelledFromRule() throws IOException {
+
+    ArabicMorfologikSpellerRule rule = new ArabicMorfologikSpellerRule(TestTools.getMessages("ar"), Languages.getLanguageForShortCode("ar"));
```

Comment:
I'm not sure, does Arabic have two spellers, hunspell and Morfologik based? Or am I missing something?

Reply:
Actually we developed both but we use only the hunspell one for now.

Reply:
The morfologik one is not considering ta3reef (https://en.wikipedia.org/wiki/Arabic_definite_article)

Reply:
removed in https://github.com/languagetool-org/languagetool/pull/2219/commits/086e90fe708eb0ec10e4f4afec28cbd0a0fbf682